src/core/Akka.Streams/Actors/ActorPublisher.cs

```diff
@@ -388,7 +388,7 @@ protected override bool AroundReceive(Receive receive, object message)
                 if (req.Count < 1)
                 {
                     if (_lifecycleState == LifecycleState.Active)
-                        OnError(new IllegalStateException("Number of requested elements must be positive"));
+                        OnError(new ArgumentException("Number of requested elements must be positive. Rule 3.9"));
```

Comment:
The TCK expects a `ArgumentException` instead of `IllegalStateException` also the error message must contain the violated rule.